### PR TITLE
Kafka error list extended

### LINKED ISSUE #42: Add more kafka error codes
Greetings!

I got error code = 19 from apache kafka and after investigation I found that not all return codes documented, but can be returned.

https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-ErrorCodes

Please, add more kafka error codes:

https://github.com/apache/kafka/blob/b62f8ea43b6d5307f7274fbe8b7984dd5ee22239/clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

### PR DESCRIPTION
**Pull request by husio**
More errors, as defined in
https://github.com/apache/kafka/blob/b62f8ea43b6d5307f7274fbe8b7984dd5ee22239/clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

closes #42

### CODE REVIEW

File: proto/errors.go [hunk 1]
```diff
@@ -5,22 +5,36 @@ import (
...
+	ErrRequestTimeout                          = &KafkaError{7, "request timeed out"}
+	ErrBrokerNotAvailable                      = &KafkaError{8, "broker not available"}
+	ErrReplicaNotAvailable                     = &KafkaError{9, "replica not available"}
+	ErrMessageSizeTooLarge                     = &KafkaError{10, "message size too large"}
+	ErrScaleControllerEpoch                    = &KafkaError{11, "scale controller epoch"}
+	ErrOffsetMetadataTooLarge                  = &KafkaError{12, "offset metadata too large"}
+	ErrOffsetLoadInProgress                    = &KafkaError{14, "offsets load in progress"}
+	ErrNoCoordinator                           = &KafkaError{15, "consumer coordinator not available"}
+	ErrNotCoordinator                          = &KafkaError{16, "not coordinator for consumer"}
+	ErrRecortListTooLarge                      = &KafkaError{17, "operation on an invalid topic"}
```
Comment (zorkian) [inline proto/errors.go]: 17 should probably be `ErrInvalidTopic`.

File: proto/errors.go [hunk 2]
```diff
@@ -5,22 +5,36 @@ import (
...
+	ErrListTooLarge                            = &KafkaError{18, "essage batch larger than the configured segment size"}
```
Comment (zorkian) [inline proto/errors.go]: Suggest `ErrRecordListTooLarge` to match the Java name.

File: proto/errors.go [hunk 3]
```diff
@@ -5,22 +5,36 @@ import (
...
+	ErrNotEnoughReplicas                       = &KafkaError{19, "not enough in-sync replicas"}
+	ErrNotEnoughReplicasAfterAppend            = &KafkaError{20, "messages are written to the log, but to fewer in-sync replicas than required"}
+	ErrInvalidRequiredAcks                     = &KafkaError{21, "invalid value for required acks"}
+	ErrIllegalGenerations                      = &KafkaError{22, "consumer generation id is not valid"}
```
Comment (zorkian) [inline proto/errors.go]: Suggest `ErrIllegalGeneration` (singular).

**Comment by husio** [comment]
F@$*, I have merged it too fast. Awesome that you noticed those bugs.

**Comment by husio** [comment]
@zorkian thanks for the check. I just pushed update to master, please have a look https://github.com/optiopay/kafka/commit/310ca4c4e54e463c4725521ca5b54513801e8834

**Comment by zorkian** [comment]
Patch LGTM. Thank you!

**Pull request merged**